### ISSUE TITLE
1.9.0 Missing Some Charts

### ISSUE THREAD

**Issue by nexusmaniac** [titpetric/netdata#64]
Hey :)

Awesome stuff here with Netdata! Been using it everywhere... But on my unRAID machine 1.9 isn't playing nice. I've dropped back to 1.8 for now.

CPU Clock (cpu.cpufreq)
coretemp-isa-0000 temperature (sensors.coretemp_isa_0000_temperature)

Are some of a small handful of charts which are non-existent on 1.9.

I'm using this docker container: https://hub.docker.com/r/titpetric/netdata/ on unRAID 6.4.1

I'm happy to share anything needed to help diagnose this :) I didn't want to paste the entire netdata.conf! It's 10k lines haha ;)

Let me know what I can do here - I'd love to be rocking the latest and greatest :D

**Comment by nexusmaniac** [comment]
```
2018-02-08 22:05:59: charts.d: INFO: cpufreq: is disabled. Add a line with cpufreq=force in /etc/netdata/charts.d.conf to enable it (or remove the line that disables it).
2018-02-08 22:05:59: charts.d: INFO: example: is disabled. Add a line with example=force in /etc/netdata/charts.d.conf to enable it (or remove the line that disables it).
2018-02-08 22:05:59: charts.d: INFO: exim: is disabled. Add a line with exim=force in /etc/netdata/charts.d.conf to enable it (or remove the line that disables it).
2018-02-08 22:05:59: charts.d: INFO: hddtemp: is disabled. Add a line with hddtemp=force in /etc/netdata/charts.d.conf to enable it (or remove the line that disables it).
2018-02-08 22:05:59: charts.d: INFO: load_average: is disabled. Add a line with load_average=force in /etc/netdata/charts.d.conf to enable it (or remove the line that disables it).
2018-02-08 22:05:59: charts.d: INFO: mem_apps: is disabled. Add a line with mem_apps=force in /etc/netdata/charts.d.conf to enable it (or remove the line that disables it).
2018-02-08 22:05:59: charts.d: INFO: mysql: is disabled. Add a line with mysql=force in /etc/netdata/charts.d.conf to enable it (or remove the line that disables it).
2018-02-08 22:05:59: node.d.plugin: ERROR: Cannot read configuration file /etc/netdata/node.d.conf: ENOENT, no such file or directory '/etc/netdata/node.d.conf', using internal defaults.

2018-02-08 22:05:59: charts.d: INFO: nginx: is disabled. Add a line with nginx=force in /etc/netdata/charts.d.conf to enable it (or remove the line that disables it).
2018-02-08 22:05:59: charts.d: INFO: phpfpm: is disabled. Add a line with phpfpm=force in /etc/netdata/charts.d.conf to enable it (or remove the line that disables it).
2018-02-08 22:05:59: charts.d: INFO: postfix: is disabled. Add a line with postfix=force in /etc/netdata/charts.d.conf to enable it (or remove the line that disables it).
2018-02-08 22:05:59: node.d.plugin: ERROR: Cannot load module's stiebeleltron config from /etc/netdata/node.d/stiebeleltron.conf exception: Error: ENOENT, no such file or directory '/etc/netdata/node.d/stiebeleltron.conf', using internal defaults.

2018-02-08 22:05:59: charts.d: INFO: sensors: is disabled. Add a line with sensors=force in /etc/netdata/charts.d.conf to enable it (or remove the line that disables it).

```

Even after running `docker exec -it NetdataL bash` and going to `/etc/netdata/` the only thing in there is an empty `override` folder. No `.conf` files to be edited...

**Comment by titpetric** [comment]
Hello,

can you paste your complete `docker run...` command. Regarding second comment: Generally most of these things will end up being disabled on docker, because the applications it monitors aren't available inside docker.

Not sure if something changed upstream between 1.9 and 1.8, but it would be worth checking it out from the official docs. Since it's a minor version change, things should be complatible enough, but maybe the official release notes might give you more information.

**Comment by nexusmaniac** [comment]
Sure thing, please see below for both run commands. Nothing's changed between 1.8 and 1.9 on my run command, all the same `dirs` are accessible via docker. Only difference here is the new Netdata version.

1.8 shows CPUfreq, full sensors, etc. 1.9 misses a few things for some reason.

**Docker run 1.8**
```
root@localhost:# /usr/local/emhttp/plugins/dynamix.docker.manager/scripts/docker run -d --name='Netdata' --net='host' -e TZ="Europe/London" -e HOST_OS="unRAID" -v '/mnt/user/appdata/netdata':'/etc/netdata':'rw' -v '/proc':'/host/proc':'ro' -v '/sys':'/host/sys':'ro' -v '/var/run/docker.sock':'/var/run/docker.sock':'ro' --cap-add SYS_PTRACE --log-opt max-size=50m --log-opt max-file=1 'titpetric/netdata:1.8'
49721c4d6112f1c0900e6c0085b783a06af2f09361d79b96a4e68da6f9e4438d

The command finished successfully!
```

**Docker Run 1.9 (Latest)**

```
root@localhost:# /usr/local/emhttp/plugins/dynamix.docker.manager/scripts/docker run -d --name='NetdataL' --net='bridge' -e TZ="Europe/London" -e HOST_OS="unRAID" -p '29999:19999/tcp' -v '/mnt/cache/appdata/netdata_latest/':'/etc/netdata':'rw' -v '/proc':'/host/proc':'ro' -v '/sys':'/host/sys':'ro' -v '/var/run/docker.sock':'/var/run/docker.sock':'ro' --cap-add SYS_PTRACE --log-opt max-size=50m --log-opt max-file=1 'titpetric/netdata:latest'
6712533310e784ebb9c958543908253d4d6267e91ac35cc74b1b9a49d99769a1

The command finished successfully!
```

I'll try looking at the Netdata changelogs, see if anything sticks out. Not sure exactly what I'll be looking for yet 😛

**Comment by ktsaou** [comment]
netdata v1.9 has a new python plugin ( rewritten by @l2isbad ). It it backwards compatible though, so I can't really tell if this introduced an issue or not.

@nexusmaniac can you post this from the v1.9 container:

```
grep python.d /var/log/netdata/error.log
```

Please restart netdata before doing so, to make sure we have a fresh copy.

**Comment by nexusmaniac** [comment]
@ktsaou 

I saw that in the changelog but because it should be backwards compatible I didn't think much of it! :)

So the command you posted doesn't work inside the container - just hangs indefinitely...
The `error.log` file is jut a symlink to `/dev/stderr`

```
2018-02-11 21:37:27: python.d INFO: plugin: main: Using python 2
2018-02-11 21:37:27: python.d ERROR: plugin: main: [Errno 2] No such file or directory: '/etc/netdata/python.d.conf'

2018-02-11 21:37:28: netdata ERROR : PLUGINSD[python.d] : read failed

2018-02-11 21:37:28: netdata ERROR : PLUGINSD[python.d] : '/usr/libexec/netdata/plugins.d/python.d.plugin' (pid 26) disconnected after 0 successful data collections (ENDs).

2018-02-11 21:37:28: netdata ERROR : PLUGINSD[python.d] : child pid 26 exited with code 1.

2018-02-11 21:37:28: netdata ERROR : PLUGINSD[python.d] : '/usr/libexec/netdata/plugins.d/python.d.plugin' (pid 26) exited with error code 1. Disabling it.

2018-02-11 21:37:28: netdata INFO : PLUGINSD[python.d] : thread with task id 25 finished
```

So I grabbed what I could from the docker logs (not entirely sure where they're coming from if not `stderr`

Perhaps the container isn't importing all the files correctly / permissions are messed up?

```
root@3468acb6a758:/usr/libexec/netdata/plugins.d# ls -lah
total 300K
drwxr-xr-x 1 root root     456 Feb 10 13:05 .
drwxr-xr-x 1 root root      62 Feb 10 13:05 ..
-rw-r--r-- 1 root root    7.3K Feb 10 13:05 README.md
-rwxr-xr-x 1 root root     110 Feb 10 13:05 alarm-email.sh
-rwxr-xr-x 1 root root     64K Feb 10 13:05 alarm-notify.sh
-rwxr-xr-x 1 root root     265 Feb 10 13:05 alarm-test.sh
-rwsr-x--- 1 root netdata  91K Feb 10 13:05 apps.plugin
-rwxr-xr-x 1 root root    5.5K Feb 10 13:05 cgroup-name.sh
-rwsr-x--- 1 root netdata  35K Feb 10 13:05 cgroup-network
-r-xr-x--- 1 root root    6.6K Feb 10 13:05 cgroup-network-helper.sh
-rwxr-xr-x 1 root root    1.2K Feb 10 13:05 charts.d.dryrun-helper.sh
-rwxr-xr-x 1 root root     20K Feb 10 13:05 charts.d.plugin
-rwxr-xr-x 1 root root    5.1K Feb 10 13:05 fping.plugin
-rw-r--r-- 1 root root    5.6K Feb 10 13:05 loopsleepms.sh.inc
-rwxr-xr-x 1 root root     11K Feb 10 13:05 node.d.plugin
-rwxr-xr-x 1 root root     13K Feb 10 13:05 python.d.plugin
-rwxr-xr-x 1 root root    6.7K Feb 10 13:05 tc-qos-helper.sh
root@3468acb6a758:/usr/libexec/netdata/plugins.d#
```

Just a guess on that last one 😄

**Comment by ktsaou** [comment]
hm... these logs are not what we need.
Can you run the plugin by hand inside the container?

You should use something like:

```sh
export NETDATA_HOST_PREFIX="/host"
/usr/libexec/netdata/plugins.d/python.d.plugin debug trace 1 cpufreq
```

**Comment by nexusmaniac** [comment]
Here you go 👍 

```
root@3468acb6a758:/# export NETDATA_HOST_PREFIX="/host"
root@3468acb6a758:/# /usr/libexec/netdata/plugins.d/python.d.plugin debug trace 1 cpufreq
2018-02-11 21:51:39: python.d INFO: plugin: main: Using python 2
2018-02-11 21:51:39: python.d ERROR: plugin: main: [Errno 2] No such file or directory: '/usr/libexec/netdata/plugins.d/../../../../etc/netdata/python.d.conf'
2018-02-11 21:51:39: python.d ERROR: plugin: main: Traceback (most recent call last):
  File "/usr/libexec/netdata/plugins.d/python.d.plugin", line 19, in <module>
    from time import monotonic as time
ImportError: cannot import name monotonic
```

**Comment by ktsaou** [comment]
oh... this seems a package dependency problem. You are missing a package (I guess it is called `monotonic` https://pypi.python.org/pypi/monotonic ).

**Comment by nexusmaniac** [comment]
@ktsaou 

It's more than a package issue (not sure the package is a blocker here, it's in a `try: except:` ...

I removed the try & except there and still it wouldn't run!

All I did to fix it was `touch /etc/netdata/python.d.conf` I'm going to restart the container to see if the dependency still causes an issue after touching the empty `python.d.conf` file

I believe that Netdata 1.9 expects lots of `.conf`'s to be in `/etc/netdata`? This container from @titpetric has nothing in that folder 😋

**Comment by nexusmaniac** [comment]
**From `/usr/libexec/netdata/plugins.d/python.d.plugin`**
```
try:
    from time import monotonic as time
except ImportError:
    from time import time
```

That try & except functions perfectly 👍 

----------------------

**Output from `/usr/libexec/netdata/plugins.d/python.d.plugin debug trace 1 cpufreq`**

```
2018-02-11 22:06:55: python.d DEBUG: cpufreq: cpufreq: update => [OK] (elapsed time: 1, retries left: 60)
2018-02-11 22:06:55: python.d DEBUG: cpufreq: cpufreq: sleeping for 0.997086048126 to reach frequency of 1 sec
BEGIN cpu.cpufreq 999983
SET 'cpu0' = 1088070
SET 'cpu1' = 880198
SET 'cpu2' = 823762
SET 'cpu3' = 924000
SET 'cpu4' = 908000
SET 'cpu5' = 1046000
SET 'cpu6' = 912000
SET 'cpu7' = 1057020
END

BEGIN netdata.runtime_cpufreq 999983
SET run_time = 1
END
```

Just having `/etc/netdata/python.d.conf` fixes this issue, I expect that the container will need to include empty `.conf` files 

Having that file there allows the web UI to render the charts!! 💯 Spot on 👍 

![image](https://user-images.githubusercontent.com/4521949/36079177-48ab16ac-0f78-11e8-8e16-8441194b23fc.png)

**Comment by ktsaou** [comment]
hm... we need @titpetric for this...

**Comment by titpetric** [comment]
TL;DR: I should install monotonic in the container? what about using python `time` as fallback as was suggested (is there some feature of monotonic netdata relies upon so stdlib isn't viable)?

**Comment by nexusmaniac** [comment]
@titpetric that's not the issue here.

The issue is the lack of .conf files in `/etc/netdata` my log stream is full of missing .conf files! There doesn't seem to be a requirement of them containing anything... 🤔 But best to talk with @ktsaou on what they're used for :)

The `time` requirement is a try/except loop which is a non-issue as per: https://github.com/titpetric/netdata/issues/64#issuecomment-364794110

TL:DR - We need `.conf` files correctly stored in `/etc/netdata` inside the container :)

Hope this helps! 😄

**Comment by titpetric** [comment]
@ktsaou at some point, netdata didn't need any conf files and they would be "generated" (basically with a list of default values) when netdata would start up. Behaviour changed recently?

**Comment by titpetric** [comment]
@nexusmaniac also if i understood correctly, if i have at least one 0 byte config in /etc/netdata, then this error doesn't show up and you get your cpufreq graphs out nicely?

**Comment by nexusmaniac** [comment]
@titpetric 

I've never had conf files generated by your Netdata container in my `/etc/netdata` DIR 😄 

------------

Correct!

I have `python.d.conf` in `/etc/netdata` which is completely empty. It allows my python.d plugins to work flawlessly 👍

**Comment by titpetric** [comment]
I just pushed a new version which issues a `touch /etc/netdata/netdata.conf` before running netdata. I appreciate if you could do a git pull and retry without some mounts under /etc/netdata, just to see if it works as expected after the change.

Should take about ~10 minutes or so before it's built on docker hub.

Edit: ugh, i hope that having an empty netdata.conf will solve it, not that `python.d.conf` is necesarry. If not, I'll just push an update.

**Comment by nexusmaniac** [comment]
I can see that it's building on Docker hub 👍 

I'll get that updated as soon as it's available and let you know!

Not sure how much impact `netdata.conf` will have? But I'll know soon enough 😄  Thanks for the support here everyone, been good trying to get to the bottom of this 😄

**Comment by nexusmaniac** [comment]
@titpetric 

`python.d ERROR: plugin: main: [Errno 2] No such file or directory: '/etc/netdata/python.d.conf'`

Sadly not, seems that my issues with cpufreq, sensors, etc. all lie with `python.d.conf` 

`Netdata.conf` doesn't seem to do anything 😢

**Comment by titpetric** [comment]
Hey, thanks and sorry for the back and forth. Hopefully the last commit will solve the issue for you ;)

**Comment by nexusmaniac** [comment]
No worries! Thanks for working with me! 😎 

Awesome stuff :) just looked at the commit 🎉

Guess that's this issue solved! Perfect ;)

**Comment by titpetric** [comment]
Don't guess, test :)

**Comment by nexusmaniac** [comment]
Hehe yeah! I've tested ;)

Latest version works as expected 😎 cheers man!

**Comment by titpetric** [comment]
Thank you.

**Comment by ktsaou** [comment]
wow!

@l2isbad check this. If `python.d.conf` is missing, python.d.plugin does not start. Ideally the plugin should still run if this file is missing. This may be a side effect of the new python.d.plugin, since the old one did run without that file.

> @ktsaou at some point, netdata didn't need any conf files and they would be "generated" (basically with a list of default values) when netdata would start up. Behaviour changed recently?

This is true, however python.d.plugin uses the config files for configuring auto-detection. So, there is no auto-detection embedded in python modules - the python modules will just use whatever is described in the config files found under `/etc/netdata/python.d/`.

**Comment by ilyam8** [comment]
> If python.d.conf is missing, python.d.plugin does not start

True

https://github.com/firehol/netdata/blob/master/plugins.d/python.d.plugin#L197-L199

**Comment by ilyam8** [comment]
fixed in https://github.com/firehol/netdata/pull/3418
